api/handlers.go

```diff
@@ -31,3 +32,123 @@ func (api *API) userHandlerGET(w http.ResponseWriter, req *http.Request, _ httpr
 	}
 	WriteJSON(w, u)
 }
+
+// userUploadsHandler returns all uploads made by the current user.
+func (api *API) userUploadsHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	sub, _, _, err := tokenFromContext(req)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	u, err := api.staticDB.UserBySub(req.Context(), sub)
+	// TODO Should we create the user in case it's not found?
```

Comment:
I would return `404` or `400`. Kind of leaning towards a `400` because I think ` `404` is at times dubious between route not found or (in this case) user not found.

Reply:
The thinking here was focused on the way this is going to be used, rather that strictly following REST rules. From REST point of view this endpoint is shady in two ways:
1. It suggest that it's going to create a user if it doesn't exist, so it combines the functionality of GET and POST.
2. It doesn't follow the standard pattern of `/resource/resource_id` because that doesn't make much sense in our use case - a user is not supposed to access the data of another user.

If we want to go strict REST, we need to change this to `/user/user_id` but that will require our code in nginx to inspect the cookies, decrypt the right one, parse the JSON, extract the id and then make the call. Given the use case, this is adding a lot of complexity without a real benefit. I would prefer to use REST as guidelines and write the code in the most convenient way for our setup, meaning that we have just `GET /user` which returns the data of the currently authenticated user (or return Unauthorized).

There is still the question of whether this GET call should also create the user if it doesn't exist. On one hand, this is unintuitive and unexpected for a GET call. On the other hand, it's very convenient, especially when you consider that with the GET call we already have all the data we need and it will simplify the flow a lot. Otherwise the logic in nginx will be this:
1. GET /user
2. If the response is 404, POST /user (with the same params) && GET /user
3. Do the rest of the work.

I don't have a strong opinion on this one. Maybe @kwypchlo can weigh in here.